Xamarin.PropertyEditing.Mac/Controls/Custom/UnfocusableTextField.cs

```diff
@@ -32,4 +32,30 @@ void SetDefaultProperties ()
 			BackgroundColor = NSColor.Clear;
 		}
 	}
+
+	internal class VerticallyCentredUnfocusableTextField : NSView
```

Comment:
It's not entirely clear why `VerticallyCentredUnfocusableTextField` doesn't derive from `UnfocusableTextField` as the name would imply, and as it seems to semantically be. Can we really not add those constraints otherwise?

Reply:
Also, without looking in too much details, I looked at `UnfocusableTextField` usage and it seems like we use it mostly as a label, and I'm not sure I saw a lot of cases where it doesn't need to be vertically centered. Even if we did, would it not be simpler to activate those constraints based on some `VerticallyCentered` boolean property on a single class, instead of having two separate controls that only differ by some minor attribute?

Reply:
We don't always want it to be vertically centred. For example in the case of the main PropertyListView, it essentially needs to be Top aligned. In the case of various controls (like Point, Rectangle etc) it isn't constrained because the view contains several controls so the Unfocusable version is used there and not this new one. We only need it to be vertically aligned when added to table rows, for now.

AFAICT it cannot be in 1 NSTextField or a subclass because NSTextField cannot centre itself, without custom drawing. Hence why it is nbow wrapped inside a NSView, where it is then centred vertically. I'm not happy with how that works, but Eric wasn't happy with my original solution of overriding the NSTextField DrawRect to position the text correctly, and he implemented this constraints solution. I'm just reusing his code, refactoring it, so that it can be used in more than 1 place.

Reply:
Fair enough, but my other suggestion for a way to manage that would be to have a unique control, using the `NSView` wrapping, but only doing the constraints if a `VerticallyCentered` property is true. If that's not practical or efficient, I'd still rename to clear the inheritance confusion, maybe to `VerticallyCenteredUnfocusableTextView`.

Reply:
Ah, Yes, that is possible. Not thinking straight.